### PR TITLE
Initial framework to allow testing of C++ code

### PR DESCRIPTION
**Pull request by daviddrysdale**
Best reviewed commit-by-commit; also fixes some build warnings along the way.

### CODE REVIEW

File: oak/server/logging_channel.h [hunk 1]
```diff
@@ -27,7 +27,8 @@ class LoggingChannelHalf final : public ChannelHalf {
   uint32_t Write(absl::Span<const char> data) override {
     std::string log_message(data.cbegin(), data.cend());
     LOG(INFO) << "LOG: " << log_message;
-  };
+    return data.size();
```
Comment (anghelcovici) [inline oak/server/logging_channel.h]: Why did this work before? Isn't this a compile error?

File: oak/server/wabt_output.cc [hunk 2]
```diff
@@ -26,7 +26,7 @@ std::ostream& operator<<(std::ostream& os, const wabt::Type& t) {
 
 std::ostream& operator<<(std::ostream& os, const std::vector<wabt::Type>& vt) {
   os << "(";
-  for (int ii = 0; ii < vt.size(); ii++) {
+  for (size_t ii = 0; ii < vt.size(); ii++) {
```
Comment (anghelcovici) [inline oak/server/wabt_output.cc]: why size_t?

File: oak/server/logging_channel.h (hunk 1 above)
Comment (daviddrysdale) [reply oak/server/logging_channel.h]: Only a compiler warning, apparently.  The return value isn't currently used, so we technically get away with the resulting undefined behaviour.

Comment (anghelcovici) [reply oak/server/logging_channel.h]: Should we treat warnings as errors? I think, at least initially this might be better

Comment (daviddrysdale) [reply oak/server/logging_channel.h]: Yeah, that would make sense (as long as we can limit it to the code that's under our control).

**Pull request merged**